### ISSUE TITLE
Load GUI style network from text

### ISSUE THREAD

**Issue by DanielSank** [qucat/qucat#43]
I'm making a Colab demonstrating qucat for me team. I'd like to be able to show normal modes of the circuit. To do this, it seems that I need to create the Network via the GUI. It's not trivial to read from a file in Colab so I'd like to be able to create a GUI-style qucat Network by reading from e.g. the string that would otherwise be in the text file created by the GUI. Is there presently a convenient way to do that? If not, I'm sure I can add something.

**Comment by mgely** [comment]
Here's a working example:
```python
filestring = """
C;-1,-2;0,-2;1.0e-13;
J;-1,-1;0,-1;1.0e-08;
W;-1,-1;-1,-2;;
W;0,-1;0,-2;;
"""
circuit = hacked_GUI(filestring)
circuit.show_normal_mode(0)
```python

**Comment by DanielSank** [comment]
How about a method like this

```python
class GUI:

    @classmethod
    def from_text(text: str) -> GUI:
```

I can make a pull request with something of that flavor.

**Comment by mgely** [comment]
That's a very elegant way of doing it! I would really appreciate that contribution thanks!

**Comment by DanielSank** [comment]
Ok, on it.

**Comment by DanielSank** [comment]
I just made a pull request to fix this issue. Unfortunately it seems I don't have edit access on that pull request so I can't leave any notes. That being the case, here are some notes:

1. It's not clear to me what the documentation standard is for this project. I'd like to add docstrings, but what format should be used?

2. I checked that the new method `qucat.core.GUI.from_text` does indeed work, but it's not clear to me how to unittest this project.

Any advice is appreciated.

**Comment by mgely** [comment]
Today I'll write a "guide for developpers" which will adress these points

**Comment by mgely** [comment]
Hi Daniel, I added a [page to the website](https://qucat.org/devs/index.html) for developers. 
This hopefully addresses your questions.

Here are links to answer your specific questions:
1. [Here](https://qucat.org/devs/comments.html) is information on the types of docstrings qucat documentation uses.

2. [Here](https://qucat.org/devs/unittests.html) is some information on how the unittesting is organized:
in your case I would append the ``tests/test_core.py`` file with a test such as
```
class CircuitInstantiation (TestCaseAppended):
    '''Test the creation of Qcircuit objects
    '''
    def test_GUI_from_string(self):
        # create simple GUI circuit from string
        # do or assert something else ?maybe?
```

Please let me know if you would like me to be more specific.

BTW, I'm not sure how to address the fact that you don't have edit access on that pull request. 
At this stage I don't actually know how to deal with giving permissions to contributors, I'll look into this tomorrow and get back to you.

**Comment by DanielSank** [comment]
Thanks, @mgely. I think it may have been a browser issue (or maybe GitHub was having a problem) because I was able to edit my comment in the pull request later.

**Comment by mgely** [comment]
Addressed in 4e52310d73a09aeb9edaa91830e3cc04e291b809
